gtsam/nonlinear/utilities.h

```diff
@@ -142,6 +143,22 @@ Matrix extractPose3(const Values& values) {
   return result;
 }
 
+/// Extract all Vector values into a single tall vector
+Vector extractVector(const Values& values) {
```

Comment:
I think an API that makes more sense is to give a character, say 'x`' that will be used to select all values with a particular Symbol, e.g. x4, x5, x6, and that will assume that they are all of the same dimension. Use case is then:
```
result = extractVectors(values, 'x')
```
and if there are 200 such vectors, and they are 5D, the result is a 200x5 Matrix (not 5x200, we want to be consistent with other methods in this file). You can thrown an exception if those values have different dimensions.